src/marks/tip.js

```diff
@@ -322,28 +307,125 @@ function getPath(anchor, m, r, width, height) {
   }
 }
 
-function getSources({channels}) {
+// Note: mutates this.format!
+function getSourceChannels(index, {channels}, scales) {
+  const {facet, format} = this;
   const sources = {};
+
+  // Prioritize channels with explicit formats, in the given order.
+  for (const key in format) {
+    const value = format[key];
+    if (value === null || value === false) {
+      // Promote x and y null shorthand for paired channels (in order).
+      if (key === "x" || key === "y") {
+        const key1 = `${key}1`;
+        const key2 = `${key}2`;
+        if (!(key1 in format)) format[key1] = null;
+        if (!(key2 in format)) format[key2] = null;
+      }
+    } else if (key === "fx" || key === "fy") {
+      sources[key] = true;
+    } else {
+      let source = getSource(channels, key);
+      if (source) sources[key] = source;
+      // Promote x and y non-null shorthand for paired channels (in order).
+      if (key === "x" || key === "y") {
+        const key1 = `${key}1`;
+        const key2 = `${key}2`;
+        if (!(key1 in format) && (source = getSource(channels, key1))) (format[key1] = value), (sources[key1] = source);
+        if (!(key2 in format) && (source = getSource(channels, key2))) (format[key2] = value), (sources[key2] = source);
```

Comment:
Ugh, this only works the first time the mark is rendered. Because we mutate `format` here, thus adding the paired channels to the end, we’re changing the order the second time around. 🤦 I need to do this in a way that puts the paired channels (_x1_ and _x2_) at the same priority as the shorthand (_x_).